feat: course backend

KindImagination opened the pull request:
Now it is possible to load courses directly from the database
closes https://github.com/we-dance/v4/issues/95

razbakov commented on cli/prisma/schema.prisma:
```diff
@@ -411,7 +410,7 @@ model Course {
 
   // Relations
-  instructor  Profile?           @relation("course_instructor", fields: [instructorId], references: [id])
+  instructor  Profile?           @relation("course_instructor", fields: [instructorUsername], references: [username])
```
here it should be instructorId

razbakov commented on components/CoursesList.vue:
```diff
@@ -113,16 +134,16 @@
                 </div>
                 <div class="text-xs text-muted-foreground">
-                  {{ course.instructor.jobTitle }}
+                  {{ course.educationalLevel }}
```
this was instructor information, now it's course?

razbakov commented on components/CoursesList.vue:
```diff
@@ -175,47 +230,71 @@ const activeFiltersCount = computed(() => {
-  const monthlyOffer = course.offers?.find((offer) => offer.duration === 'P1M')
-  if (monthlyOffer) {
-    return `${monthlyOffer.price} ${monthlyOffer.priceCurrency}`
+const getMonthlyPrice = (course: any) => {
```
so it shows only price for courses with monthly subscription?

KindImagination committed:
courses list API

KindImagination committed:
refactor: update pricing logic in CoursesList component

KindImagination committed:
feat: add instructor bio to CoursesList component and API response

- Updated CoursesList.vue to display instructor bio instead of educational level.
- Modified courses API to include instructor bio in the response.
